analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+
+```{r}
+# get matrix of barcodes by chromosome
+gain_cnv_df <- coldata_df |> 
+  dplyr::select(barcodes, starts_with("proportion_scaled_dupli")) |> 
+  unique() |> 
+  tibble::column_to_rownames("barcodes") |> 
+  as.matrix()
+
+# remove extra characters in colnames, proportion_scaled_dupli
+colnames(gain_cnv_df) <- stringr::word(colnames(gain_cnv_df), -1, sep = "_")
+
+# need to make a new annotation, since we want to annotate the rows here 
+annotation <- ComplexHeatmap::rowAnnotation(classification = annotation_df$combined_cnv_classification,
+                                            col = list(classification = c("Tumor" = "#00274C", "Normal" = "#FFCB05")))
+```
+
+
+```{r}
+ComplexHeatmap::Heatmap(
+  gain_cnv_df,
+  col = circlize::colorRamp2(c(0, 1), colors = c("white", "#00274C")),
+  border = TRUE,
+  ## Row parameters
+  cluster_rows = TRUE,
+  row_title = "",
+  row_title_gp = grid::gpar(fontsize = 12),
+  row_title_side = "left",
+  show_row_names = FALSE,
+  row_dend_side = "right",
+  row_names_gp = grid::gpar(fontsize = 10),
+  ## Column parameters
+  cluster_columns = FALSE,
+  show_column_names = TRUE,
+  right_annotation = annotation
+)
+```
+
+Here we see a gain in Chr8 and Chr1 in tumor cells compared to normal cells. 
+We also see other chromosomes with amplifications in tumor cells compared to normal cells and that the tumor and normal cells are nicely separated in the heirarchical clustering. 
+
+Below we make the same plot but with only the losses or deletions. 
+
+```{r}
+# barcode by chromosome mtx with losses 
+loss_cnv_df <- coldata_df |> 
+  dplyr::select(barcodes, starts_with("proportion_scaled_loss")) |> 
+  unique() |> 
+  tibble::column_to_rownames("barcodes") |> 
+  as.matrix()
+  
+colnames(loss_cnv_df) <- stringr::word(colnames(loss_cnv_df), -1, sep = "_")
+```
+
+
+```{r}
+ComplexHeatmap::Heatmap(
+  loss_cnv_df,
+  col = circlize::colorRamp2(c(0, 1), colors = c("white", "#00274C")),
+  border = TRUE,
+  ## Row parameters
+  cluster_rows = TRUE,
+  row_title = "",
+  row_title_gp = grid::gpar(fontsize = 12),
+  row_title_side = "left",
+  show_row_names = FALSE,
+  row_dend_side = "right",
+  row_names_gp = grid::gpar(fontsize = 10),
+  ## Column parameters
+  cluster_columns = FALSE,
+  show_column_names = TRUE,
+  right_annotation = annotation
+)
+```
+
+We do not see the expected loss in Chr16, but this is not present in every patient. 
+Additionally, we had already noted the gain in Chr8 in the tumor cells, so there is still at least one expected CNV present. 
+Again, there is a clear separation of tumor and normal cells in terms of presence of CNVs. 
+
+## Save final tumor annotations
+
+After validating that the cells classified as tumor by both `CopyKAT` and `InferCNV` show the expected expression patterns and CNV patterns as tumor cells, we can confidently label those cells as tumor cells. 
```

Comment:
I think we might want to be a bit more careful about the final annotations here. I agree that the "Tumor" annotations seem reliable, but I am less confident in the "Normal", so I think we may want to retain some information about "unknown" or "unclassified" cells.